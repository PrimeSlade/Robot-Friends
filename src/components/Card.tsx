import React from "react";

type CardProp = {
  id: number;
  name: string;
  email: string;
};

const Card = ({ id, name, email }: CardProp) => {
  return (
    <div
      className="bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5"
      data-testid="card"
    >
      <img src={`https://robohash.org/${id}?200*200`} alt="robot" />
      <div>
        <h2>{name}</h2>
        <p>{email}</p>
      </div>
    </div>
  );
};

export default Card;
