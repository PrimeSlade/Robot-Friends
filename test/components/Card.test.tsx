import { render } from "@testing-library/react";
import Card from "../../src/components/Card";
import React from "react";

test("testing card", () => {
  const { asFragment } = render(
    <Card id={undefined} name={undefined} email={undefined} />
  );
  expect(asFragment()).toMatchSnapshot();
});
