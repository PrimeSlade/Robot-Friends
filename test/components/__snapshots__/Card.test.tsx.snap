// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing card 1`] = `
<DocumentFragment>
  <div
    class="bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5"
    data-testid="card"
  >
    <img
      alt="robot"
      src="https://robohash.org/undefined?200*200"
    />
    <div>
      <h2 />
      <p />
    </div>
  </div>
</DocumentFragment>
`;
